### PR TITLE
Feature: Added an out-of-proc WinRT server for long-running background tasks

### PR DESCRIPTION
**Pull request by hez2010**
**Changes**
- Upgraded to .NET 8
- Applied optimizations to all projects
- Added an out-of-proc WinRT server for long-running background tasks
- Added an `AppInstanceMonitor` to exit the server automatically if no Files app instance exists

Classes and delegates are supported, to add a class or delegate in the WinRT server:
1. It must be sealed if it's a class. 
2. Make sure you also add the type as an `ActivatableClass` in Package.appxmanifest under `OutOfProcessServer`. 
3. Methods, properties and events for both static and non-static are supported, but the type must have a projection (primitive types, types that have a [.NET/WinRT mapping](https://learn.microsoft.com/en-us/windows/apps/develop/platform/csharp-winrt/net-mappings-of-winrt-types), WinRT types and types defined in the server project) before it can be used in the signature. 

Notes:
The WinRT server is an RPC server and it **doesn't** own the object being passed to it from the client, instead it only holds a reference to it. Although you can pass an object from client to the server and use it in the server, you cannot close the client in this case as the object reference will be invalid if the client gone. If you want to use an object even after closing the client or want to share it between clients, the best approach is to instantiate the object on the server and pass it to the client so that the object will live as long as the server lives (the server won't be closed once launched, unless it was terminated manually). 

Contributes to #13632 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App (Package)/Package.appxmanifest [hunk 1]
```diff
@@ -160,6 +160,19 @@
                 <ActivatableClass ActivatableClassId="Files.App.BackgroundTasks.UpdateTask" ThreadingModel="both" />
             </InProcessServer>
         </Extension>
+        <Extension Category="windows.activatableClass.inProcessServer">
+            <InProcessServer>
+                <Path>Files.App\Files.App.Server.dll</Path>
+                <ActivatableClass ActivatableClassId="Files.App.Server.CsWinRTWorkaround" ThreadingModel="both" />
```
Comment (hez2010) [inline src/Files.App (Package)/Package.appxmanifest:166]: This line is necessary to workaround a bug in the appx manifest.

File: src/Files.App.Server/Files.App.Server.csproj [hunk 2]
```diff
@@ -32,9 +37,16 @@
...
     <ItemGroup>
         <Manifest Include="app.manifest" />
-        <TrimmerRootAssembly Include="Files.App.Server"></TrimmerRootAssembly>
+        <TrimmerRootAssembly Include="Files.App.Server" />
         <PackageReference Include="Microsoft.Windows.CsWinRT" Version="2.0.4" />
         <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.1.647-beta" PrivateAssets="all" />
     </ItemGroup>
 
+    <ItemGroup>
+        <None Include="..\Files.App.Server\bin\$(Platform)\$(Configuration)\$(TargetFramework)\win-$(Platform)\Files.App.Server.winmd">
```
Comment (hez2010) [inline src/Files.App.Server/Files.App.Server.csproj:46]: A workaround for appx toolchains. Without this the implementation of the server (Files.App.Server.dll) will be removed from the package layout.

File: src/Files.App/nupkgs/Microsoft.UI.Content.Private/WinRT.cs
Comment (yaira2) [inline src/Files.App/nupkgs/Microsoft.UI.Content.Private/WinRT.cs:1]: Why was this removed?

Comment (yaira2) [reply src/Files.App/nupkgs/Microsoft.UI.Content.Private/WinRT.cs:1]: We still need this for shell previews https://github.com/files-community/Files/pull/13096

Comment (hez2010) [reply src/Files.App/nupkgs/Microsoft.UI.Content.Private/WinRT.cs:1]: This file will be auto-generated by CsWinRT introduced by this PR.

Comment (yaira2) [reply src/Files.App/nupkgs/Microsoft.UI.Content.Private/WinRT.cs:1]: I believe it had some slight modifications so it might need to be restored.  You can confirm by trying to preview a word document and seeing if the app crashes.

Comment (hez2010) [reply src/Files.App/nupkgs/Microsoft.UI.Content.Private/WinRT.cs:1]: I pushed a new commit that resolved this issue.

**Review by yaira2 — APPROVED** [review]
LGTM
